samples/Bundling/rollup/js/rollup.config.js

```diff
@@ -0,0 +1,39 @@
+import resolve from "rollup-plugin-node-resolve";
+import cjs from "rollup-plugin-commonjs";
+import json from "rollup-plugin-json";
+import shim from "rollup-plugin-shim";
+
+export default {
+  input: "src/index.js",
+  output: {
+    file: "dist/bundle.js",
+    format: "iife",
+    name: "main"
+  },
+  plugins: [
+    shim({
```

Comment:
I'm curious, does rollup read the `browser` field in the package.json of whatever is being bundled?

I ask because things like [fs](https://github.com/Azure/azure-sdk-for-js/blob/662738385d36984a7135e710ec25af6589bfab1b/sdk/storage/storage-blob/package.json#L14) should be shimmed according to the mappings if rollup does. Otherwise...can we realistically recommend using rollup if the user has to know what each package depends on to shim appropriately?

Reply:
rollup doesn't seem to do any magic on its own. @bterlson already filed an issue to make this less painful: https://github.com/Azure/azure-sdk-for-js/issues/5951

Reply:
I wouldn't say we're recommending rollup, just recognizing that many people use it and giving instructions on how to do it.

Aside from #5891 I agree that the package.json browser map should be handling this, but Jeff found it wasn't working. I think this is fine for now, and I will investigate whether there's some config that will make the browser map work. It might also be a rollup bug of some kind...

Reply:
Gotcha. Do we also need the `isNode` replacement config?

Reply:
In-person chat about this: we decided we'd just fix this in our code to do the right thing